src/main/scala/com/cognite/sdk/scala/v1/resources/files.scala

```diff
@@ -24,24 +30,58 @@ class Files[F[_]](val requestSession: RequestSession[F])
 
   implicit val errorOrFileDecoder: Decoder[Either[CdpApiError, File]] =
     EitherDecoder.eitherDecoder[CdpApiError, File]
+
+  override def createOne(item: FileCreate): F[File] =
+    requestSession
+      .send { request =>
+        request
+          .post(baseUri)
+          .body(item)
+          .response(asJson[Either[CdpApiError, File]])
+          .mapResponse {
+            case Left(value) => throw value.error
+            case Right(Left(cdpApiError)) =>
+              throw cdpApiError.asException(uri"$baseUri/byids")
+            case Right(Right(value)) => value
+          }
+      }
+
   override def createItems(items: Items[FileCreate]): F[Seq[File]] =
     items.items match {
-      case item :: Nil =>
-        requestSession
-          .send { request =>
-            request
-              .post(baseUri)
-              .body(item)
-              .response(asJson[Either[CdpApiError, File]])
-              .mapResponse {
-                case Left(value) => throw value.error
-                case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/byids")
-                case Right(Right(value)) => Seq(value)
-              }
-          }
+      case item :: Nil => {
+        createOne(item).map(Seq(_))
+      }
       case _ => throw new RuntimeException("Files only support creating one file per call")
     }
 
+  def upload(file: java.io.File): F[Either[String, File]] = {
+    implicit val sttpBackend: SttpBackend[Id, Nothing] = HttpURLConnectionBackend(
```

Comment:
We should be using the backend in `requestSession`. I want us to avoid adding constraints unless we absolutely have to.

`RequestSession` has a `.map` method that allows you to transform the return value "within" `F`, so we don't need constraints like `: Functor` on resource types (instead keeping it isolated to `RequestSession`).

Reply:
It's possible that's not the best way to do it and it is a bit awkward, but for now I'd like us to stick with that pattern.